chain/beacon/node.go

```diff
@@ -325,7 +325,7 @@ func (h *Handler) run(startTime int64) {
 				// XXX find a way to start the catchup as soon as the runsync is
 				// done. Not critical but leads to faster network recovery.
 				h.l.Debugw("", "beacon_loop", "run_sync_catchup", "last_is", lastBeacon, "should_be", current.round)
-				go h.chain.RunSync(context.Background(), current.round, nil)
+				h.chain.RunSync(current.round, nil)
```

Comment:
do we want to block here? - isn't this potentially going to be a long operation?

Reply:
It's fine to block here because now `chain.RunSync` is just sending a "Sync Request" to the Sync Manager, instead of doing the actual work, and it's the Sync Manager that will do the work in its own go routine.